packages/react-core/src/demos/ComposableMenu/examples/ComposableTypeaheadSelect.tsx

```diff
@@ -100,33 +108,41 @@ export const ComposableTypeaheadSelect: React.FunctionComponent = () => {
     switch (event.key) {
       // Select the first available option
       case 'Enter':
-        if (isMenuOpen) {
+        // Only allow selection if the first item is a valid, selectable option
+        if (isMenuOpen && focusedItem.itemId !== 'no results') {
           setInputValue(String(focusedItem.children));
           setIsSelected(true);
         }
 
         setIsMenuOpen(prevIsOpen => !prevIsOpen);
         setFocusedItemIndex(null);
+        setActiveItem(null);
         focusOnInput();
 
         break;
       case 'Tab':
       case 'Escape':
         setIsMenuOpen(false);
+        setActiveItem(null);
         break;
       case 'ArrowUp':
       case 'ArrowDown':
+        event.preventDefault();
         handleMenuArrowKeys(event.key);
         break;
       default:
         !isMenuOpen && setIsMenuOpen(true);
```

Comment:
Something I did locally based on other implementations of this sort of component, is I removed this default case for the switch and replaced it with the following outside the switch:

```
if (/^(\w|\s)?$/.test(event.key) && !isMenuOpen) {
      setIsMenuOpen(true);
}
```

Something like this would only cause the menu to expand when a valid key is pressed (alphanumeric or Space). Right now *any* key will trigger the expansion, so Shift, Ctrl, etc.

Reply:
makes sense to me, seems slightly odd if it pops open when they hit shift before typing a letter. capture group isn't necessary though and i don't think it needs to allow empty (`?` makes it optional).. probably just `/^[\w ]$/` would be fine. although would we want to allow symbols and quotes and stuff? could maybe also just test `event.key.length === 1`

also, would we want to open for Backspace/Delete as well?

Reply:
Ah yeah there's non alphanumeric/space keys it should open for. Maybe rather than handling the menu opening when the input is typed into here, we add a conditional to the useEffect on line 38:

```
if (!isMenuOpen) {
  setIsMenuOpen(true)
}
```

Then only actual typing in the input would open it. Pressing backspace/delete in an empty input won't open it, but that could be considered one expected behavior.

Reply:
would `onTextInputChange` make sense?

then totally remove the open menu logic from keydown?